File: SchoolWebApp-Client/src/app/cbe/community-service/models/student-community-service-activity.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class StudentCommunityServiceActivity extends ResourceModel<StudentCommunityServiceActivity> {
    public studentId?: number;
    public student?: any;
    public communityServiceActivityId?: number;
    public communityServiceActivity?: any;
    public sessionId?: number;
    public session?: any;
    public academicYearId?: number;
    public academicYear?: any;
    public description?: string;

    constructor(model?: Partial<StudentCommunityServiceActivity>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/CBE/CommunityService/StudentCommunityServiceActivity.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;
using SchoolWebApp.Core.Entities.Students;

namespace SchoolWebApp.Core.Entities.CBE.CommunityService
{
    public class StudentCommunityServiceActivity : Base
    {
        public int StudentId { get; set; }
        public Student? Student { get; set; } = null;
        public int CommunityServiceActivityId { get; set; }
        public CommunityServiceActivity? CommunityServiceActivity { get; set; } = null;
        public int SessionId { get; set; }
        public Session? Session { get; set; } = null;
        public int AcademicYearId { get; set; }
        public AcademicYear? AcademicYear { get; set; } = null;
        public string? Description { get; set; } = null;
    }
}
